airflow/providers/google/cloud/operators/workflows.py

```diff
@@ -0,0 +1,638 @@
+        self.log.info("Retrieving workflow")
+        workflow = hook.get_workflow(
+            workflow_id=self.workflow_id,
+            location=self.location,
+            project_id=self.project_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        return Workflow.to_dict(workflow)
+
+
+class WorkflowExecutionsCreateExecutionOperator(BaseOperator):
+    """
+    Creates a new execution using the latest revision of
+    the given workflow.
+
+    :param execution: Required. Execution to be created.
+    :type execution: Dict
+    :param workflow_id: Required. The ID of the workflow.
+    :type workflow_id: str
+    :param project_id: Required. The ID of the Google Cloud project the cluster belongs to.
+    :type project_id: str
+    :param location: Required. The Cloud Dataproc region in which to handle the request.
+    :type location: str
+    :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+        retried.
+    :type retry: google.api_core.retry.Retry
+    :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+        ``retry`` is specified, the timeout applies to each individual attempt.
+    :type timeout: float
+    :param metadata: Additional metadata that is provided to the method.
+    :type metadata: Sequence[Tuple[str, str]]
+    """
+
+    template_fields = ("location", "workflow_id", "execution")
+    template_fields_renderers = {"execution": "json"}
+
+    def __init__(
+        self,
+        *,
+        workflow_id: str,
+        execution: Dict,
+        location: str,
+        project_id: str,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.workflow_id = workflow_id
+        self.execution = execution
+        self.location = location
+        self.project_id = project_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context):
+        hook = WorkflowsHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        self.log.info("Creating execution")
+        execution = hook.create_execution(
+            workflow_id=self.workflow_id,
+            execution=self.execution,
+            location=self.location,
+            project_id=self.project_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        return Execution.to_dict(execution)
```

Comment:
The code should return only after the Execution isn't in "ACTIVE" state anymore:
https://cloud.google.com/workflows/docs/reference/executions/rest/v1beta/projects.locations.workflows.executions#State

Reply:
This operator only triggers an execution. Users should use sensor to wait for execution completion. Of course we can add synchronous waiting but this may influence idempotency

Reply:
Ok, non-blocking WorkflowExecutionsCreateExecutionOperator + sensor sound good.